app/src/main/java/com/es/findsoccerplayers/LoginActivity.java

```diff
@@ -29,164 +32,147 @@
+    private static final int RC_GOOGLE_SIGN_IN = 100;
     private static final String TAG = "LoginActivity";
-    private EditText logPsw;
-    private EditText logEmail;
-    private GoogleSignInClient mGoogleSignInClient;
-    private FirebaseAuth mAuth;
-    private FirebaseUser acct;
+    private FirebaseAuth auth;
     private long backPressedTime;
     private Toast backToast;
-    private ProgressBar prgBar;
+    private ProgressBar progressBar;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        Log.w(TAG, "LoginActivity creata");
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
-        Log.w(TAG,"Activity Creata");
-        mAuth = FirebaseAuth.getInstance();
-        acct = mAuth.getCurrentUser();
+        Toolbar toolbar = findViewById(R.id.log_toolbar);
+        setSupportActionBar(toolbar);
 
-        //Se c'è gia un utente loggato, lancio MainActivity
-        if(acct!=null) {
-            startActivity(new Intent(getApplicationContext(),MainActivity.class));
+        auth = FirebaseAuth.getInstance();
+        FirebaseUser acct = auth.getCurrentUser();
+        //if someone has already logged, start MainActivity
+        if(acct !=null) {
+            startActivity(new Intent(LoginActivity.this, MainActivity.class));
             finish();
+            return;
         }
-        SignInButton btnGoogle = findViewById(R.id.googleSignInBtn);
+
+        //set the text of sign in google button to common long
+        SignInButton btnGoogle = findViewById(R.id.log_googleSignInBtn);
         btnGoogle.setSize(SignInButton.SIZE_STANDARD);
-        logEmail = findViewById(R.id.logEmail);
-        logPsw = findViewById(R.id.logPsw);
-        Button btnRegister = findViewById(R.id.btnRegister);
-        prgBar = findViewById(R.id.progBar);
-        Button btnLogin = findViewById(R.id.btnLogin);
-        Toolbar toolbar = findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
-        prgBar.setVisibility(View.INVISIBLE);
+        ((TextView)btnGoogle.getChildAt(0)).setText(R.string.common_signin_button_text_long);
+
+        final EditText logEmail = findViewById(R.id.log_email);
+        final EditText logPsw = findViewById(R.id.log_pass);
+        progressBar = findViewById(R.id.log_progressBar);
+        progressBar.setVisibility(View.INVISIBLE);
+        Button btnRegister = findViewById(R.id.log_registerBtn);
+        Button btnLogin = findViewById(R.id.log_loginBtn);
 
-        //Creo l'oggetto gso che tramite il parametro DEFAULT_SIGN_IN mi permette di creare un oggetto
-        //mGoogleSignInClient nel quale vengono memorizzate le informazioni di base dell'utente google
-        //loggato
+        //options required to log in with Google
         GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                 .requestIdToken(getString(R.string.default_web_client_id))
```

Comment:
Aggiungi nel string.xml la default_web_client_id perche non esiste

Reply:
prova a fare un sync se te la segna in rosso. Esiste, è definita da Android